### ISSUE TITLE
`uv pip install llama-cpp-python` fails "File exists when symlinking..."

### ISSUE THREAD

**Issue by abetlen** [abetlen/llama-cpp-python#1670]
# Steps

```
uv pip install llama-cpp-python
```

# Output

```
Failed to download and build `llama-cpp-python==0.2.87`
  Caused by: Failed to extract archive
  Caused by: failed to unpack `/home/andrei/.cache/uv/built-wheels-v3/.tmp5H3yE1/llama_cpp_python-0.2.87/vendor/llama.cpp/spm-headers/ggml-alloc.h`
  Caused by: File exists (os error 17) when symlinking ../ggml/include/ggml-alloc.h to /home/andrei/.cache/uv/built-wheels-v3/.tmp5H3yE1/llama_cpp_python-0.2.87/vendor/llama.cpp/spm-headers/ggml-alloc.h
```
```

**Comment by charliermarsh** [comment]
Hey @abetlen -- I didn't look into this deeply but I did notice that when I untar the latest source distribution, every entry is repeated (unlike in prior releases):

```
❯ tar -tf llama_cpp_python-0.2.87.tar.gz | tail
llama_cpp_python-0.2.87/vendor/llama.cpp/tests/test-tokenizer-0.py
llama_cpp_python-0.2.87/vendor/llama.cpp/tests/test-tokenizer-0.sh
llama_cpp_python-0.2.87/vendor/llama.cpp/tests/test-tokenizer-0.sh
llama_cpp_python-0.2.87/vendor/llama.cpp/tests/test-tokenizer-1-bpe.cpp
llama_cpp_python-0.2.87/vendor/llama.cpp/tests/test-tokenizer-1-bpe.cpp
llama_cpp_python-0.2.87/vendor/llama.cpp/tests/test-tokenizer-1-spm.cpp
llama_cpp_python-0.2.87/vendor/llama.cpp/tests/test-tokenizer-1-spm.cpp
llama_cpp_python-0.2.87/vendor/llama.cpp/tests/test-tokenizer-random.py
llama_cpp_python-0.2.87/vendor/llama.cpp/tests/test-tokenizer-random.py
llama_cpp_python-0.2.87/PKG-INFO
```

So when uv sees the repeated symlink, it tries to write it, but the target already exists.

**Comment by abetlen** [comment]
Hey @charliermarsh thank you so much for taking a look, that helped me narrow down the issue to a bug in `scikit-build-core>=0.10.0` that included files twice in the source distribution if they were explicitly included in the `pyproject.toml`.

**Comment by charliermarsh** [comment]
Awesome, I'm glad I could help!

**Comment by henryiii** [comment]
scikit-build-core 0.10.2 should fix this (thanks to @abetlen)!

**Comment by alonsosilvaallende** [comment]
This is solved in version v0.2.88.

**Comment by abetlen** [comment]
@henryiii thanks for the quick merge!
